File: coding_interview_pattern/two_pointers/python/remove_element.py
#!/usr/bin/python3
"""remove occurence of value in array
then return the new length
"""
from typing import List

def removeElement(arr: List[int], n: int, val: int) -> int:
    newLen = 0

    for i in range(n):
        if arr[i] != val:
            arr[newLen] = arr[i]
            newLen += 1
    return newLen



if __name__ == "__main__":
    arr = [4,7,6,7,8,7,7]
    size = len(arr)
    val = 7

    print(arr)

    res = removeElement(arr, size, val)
    print(res)

    print(arr[:res])

File: coding_interview_pattern/two_pointers/c/remove_element.c
#include <stdio.h>

/**
 * removeElement - remove occurence of val in array
 *
 * @arr: array
 * @n: size of the array
 * @val: occurrence of value
 * Return - the new length
 */
int removeElement(int arr[], int n, int val)
{
    int i, newLen = 0;
    for (i = 0; i < n; i++)
    {
        if (arr[i] != val)
        {
            arr[newLen++] = arr[i];
        }
    }
    return newLen;
}


int main(void)
{
    int arr[] = {1,5,5,3,5};
    int val = 5;
    int size = sizeof(arr) / sizeof(arr[0]);

    int newLen = removeElement(arr, size, val);

    printf("%d\n", newLen);

    // for (int i = 0; i < newLen; i++)
    //     printf("%d ", arr[i]);
    // printf("\n");

    return (0);
}
